engine/api/v2_workflow_run.go

```diff
@@ -0,0 +1,131 @@
+	return service.RBAC(api.workflowExecute),
+		func(ctx context.Context, w http.ResponseWriter, req *http.Request) error {
+			vars := mux.Vars(req)
+			pKey := vars["projectKey"]
+			vcsIdentifier, err := url.PathUnescape(vars["vcsIdentifier"])
+			if err != nil {
+				return sdk.NewError(sdk.ErrWrongRequest, err)
+			}
+			repositoryIdentifier, err := url.PathUnescape(vars["repositoryIdentifier"])
+			if err != nil {
+				return sdk.WithStack(err)
+			}
+			workflowName := vars["workflowName"]
+			branch := QueryString(req, "branch")
+
+			proj, err := project.Load(ctx, api.mustDB(), pKey)
+			if err != nil {
+				return err
+			}
+
+			vcsProject, err := api.getVCSByIdentifier(ctx, pKey, vcsIdentifier)
+			if err != nil {
+				return err
+			}
+
+			repo, err := api.getRepositoryByIdentifier(ctx, vcsProject.ID, repositoryIdentifier)
+			if err != nil {
+				return err
+			}
+
+			if branch == "" {
+				tx, err := api.mustDB().Begin()
+				if err != nil {
+					return err
+				}
+				vcsClient, err := repositoriesmanager.AuthorizedClient(ctx, tx, api.Cache, proj.Key, vcsProject.Name)
+				if err != nil {
+					_ = tx.Rollback()
+					return err
+				}
+				defaultBranch, err := vcsClient.Branch(ctx, repo.Name, sdk.VCSBranchFilters{Default: true})
+				if err != nil {
+					_ = tx.Rollback()
+					return err
+				}
+				if err := tx.Commit(); err != nil {
+					_ = tx.Rollback()
+					return err
+				}
+				branch = defaultBranch.DisplayID
+			}
+
+			workflowEntity, err := entity.LoadByBranchTypeName(ctx, api.mustDB(), repo.ID, branch, sdk.EntityTypeWorkflow, workflowName)
+			if err != nil {
+				return err
+			}
+
+			var wk sdk.V2Workflow
+			if err := yaml.Unmarshal([]byte(workflowEntity.Data), &wk); err != nil {
+				return err
+			}
+
+			u := getUserConsumer(ctx)
+
+			wrNumber, err := workflow_v2.WorkflowRunNextNumber(api.mustDB(), repo.ID, wk.Name)
+			if err != nil {
+				return err
+			}
+
+			wr := sdk.V2WorkflowRun{
+				ProjectKey:   proj.Key,
+				VCSServerID:  vcsProject.ID,
+				RepositoryID: repo.ID,
+				WorkflowName: wk.Name,
+				WorkflowRef:  workflowEntity.Branch,
+				WorkflowSha:  workflowEntity.Commit,
+				Status:       sdk.StatusWorkflowRunCrafting,
+				RunNumber:    wrNumber,
+				RunAttempt:   0,
+				Started:      time.Now(),
+				LastModified: time.Now(),
+				ToDelete:     false,
+				WorkflowData: sdk.V2WorkflowRunData{Workflow: wk},
+				UserID:       u.AuthConsumerUser.AuthentifiedUserID,
+				Username:     u.AuthConsumerUser.AuthentifiedUser.Username,
+				Event:        sdk.V2WorkflowRunEvent{},
+			}
+
+			tx, err := api.mustDB().Begin()
+			if err != nil {
+				return sdk.WithStack(err)
+			}
+
+			wr.RunNumber = wrNumber
+			if err := workflow_v2.InsertRun(ctx, tx, &wr); err != nil {
+				return err
+			}
+
+			select {
+			case api.workflowRunCraftChan <- wr.ID:
+				log.Debug(ctx, "postWorkflowRunV2Handler: workflow run %s %d sent into chan", wr.WorkflowName, wr.RunNumber)
+			default:
```

Comment:
i guess we can have an empty "default" section because there is a goroutine that collects from the database ?
can you just a comment ?